src/Goteo/Controller/Api/ProjectsApiController.php

```diff
@@ -625,4 +627,88 @@ public function projectDeleteSupportMsgAction($mid) {
         return $this->jsonResponse(['invest' => $message->invest]);
     }
 
+    public function projectsFootprintsOrSDGs(Request $request)
+    {
+
+        if ($request->query->has('view'))
+            $view = $request->query->get('view');
+
+        $projects = [];
+        $filters = [];
+        
+        $page = 0;
+        $offset = 0;
+        $limit = 10;
+
+        if ($request->query->has('page'))
+            $page = $request->query->get('page');
+
+        if ($request->query->has('limit'))
+            $limit = $request->query->get('limit');
+
+        if ($request->query->has('sdg') && $request->query->get('sdg')) {
+            $sdgs = explode(',', $request->query->get('sdg'));
+            $filters['sdgs'] = $sdgs;
+        }
+
+        if ($request->query->has('channel')) {
+            $filters['channel'] = $request->query->get('channel');
+        }
+
+        $total = Project::getByFootprintOrSDGs($filters, 0, 0, true);
+        $projects = Project::getByFootprintOrSDGs($filters, $page * $limit, $limit);
+
+        View::setTheme('responsive');
+
+        if ($view == 'list_projects') {
+            $response = [
+                'total' => $total,
+                'page' => $page,
+                'limit' => $limit,
+                'result_total' => count($projects),
+                'html' => View::render( 
+                    'impact_discover/partials/list_project_rows', [
+                        'projects' => $projects
+                    ]
+                )
+            ];
+        }
```

Comment:
@davidbeig I think we could have a when instead of if / else if here. Followed by calling 3 different methods that return the response.

It would be much simpler to read and to understand that way.